### PR TITLE
Refactor integration test framework and relieve the pain of regeneration

### PR DESCRIPTION
**Pull request by li-boxuan**
Integration tests can test prompts, actions, and sandboxes, to prevent the system from breaking, but they are sometimes annoying - imagine you would have to run `tests/integration/regenerate.sh` with real money cost when you only fix a typo in the prompt. They can also be annoying to reviewers because responses from LLMs can vary from every run, and thus cause a lot of diffs in the PR.

@rbren proposed "regression testing". The idea is to check-in all prompts and at the same time, not send any prompt to real LLM. This is great because it's very stable and can capture regressions in prompts, but it also has two drawbacks: 1) we have to modify OpenDevin code to set up an artificial `state` for testing purpose; 2) it no more tests action/sandbox pairs or E2E workflow.

This PR is a compromise between regression testing and the status-quo. The idea is simple: whenever `FORCE_APPLY_PROMPTS=true`, integration tests will replace existing `prompt*.log` files with newly generated prompts, and continue to use existing `response*.log` to resume the control flow. This is the default behavior now in `regenerate.sh`, although one can use `FORCE_USE_LLM=true` to switch back to the original behavior - calling LLM to generate real prompt-response pairs and check-in those log files.

The largest benefit of this PR is that one no longer needs to use real money to call LLMs and regenerate test artifacts, when they only make small changes to prompts. This PR contains an example: I updated `planner_agent`'s prompt, but only the prompt log files are updated without touching response files. No call to LLMs was made.

This is a compromise because we still lose some ability to detect bugs. With this PR, you can introduce a huge performance regression to prompts without noticing because the tests are still passing. Luckily, we now have `FORCE_USE_LLM` mode. For demo purpose, I updated `monologue_agent`'s prompt, and force the response files to be regenerated by running `FORCE_USE_LLM=true ./tests/integration/regenerate.sh`. This is useful when you, as an author or reviewer, think the changes might affect how LLMs behave (in either good or bad way).

Finally, this PR also contains a few small fixes and enhancements, including: 1) masking WORKSPACE_MOUNT_PATH_IN_SANDBOX` variable, thanks to @enyst 's finding in #1771 , 2) prettify and reorganize the code in regenerate.sh, 3) disable warnings captured by pytest.

### CODE REVIEW

File: tests/integration/conftest.py [hunk 1]
```diff
@@ -105,6 +136,8 @@ def patch_completion(monkeypatch, request):
 
 
 def set_up():
+    global cur_id
+    cur_id = 0
```
Comment (yufansong) [inline tests/integration/conftest.py:140]: Why set a global `cur_id`

**Review by rbren — APPROVED** [review]
This seems great to me!

File: tests/integration/conftest.py (hunk 1 above)
Comment (li-boxuan) [reply tests/integration/conftest.py:140]: This is stolen from an old pr from @rbren 

The idea is that the logs are in sequential order - 001.log, 002.log, etc.

By maintaining an incremental global id, we can directly look up the corresponding response log file.

Comment (yufansong) [reply tests/integration/conftest.py:140]: Get it

**Review by yufansong — APPROVED** [review]
LGTM

**Pull request merged**